frontend/components/calendar/editEvent.tsx

```diff
@@ -1,68 +1,73 @@
-import { ChangeEvent, useState } from "react";
+import React, {useState} from "react";
 import Modal from "react-bootstrap/Modal";
 import Button from "react-bootstrap/Button";
-import { User } from "@/lib/user";
-import { Tour } from "@/lib/tour";
+import StudentAutocomplete from "@/components/autocompleteComponents/studentAutocomplete";
+import TourAutocomplete from "@/components/autocompleteComponents/tourAutocomplete";
+import {handleError} from "@/lib/error";
+import ErrorMessageAlert from "@/components/errorMessageAlert";
+import {patchStudentOnTour} from "@/lib/student-on-tour";
+import {formatDate} from "@/lib/date";
 
 function EditEventModal(data: any) {
-    const { event, allStudents, allTours, isOpen, onClose, onSave, onDelete } = data;
-    const [tour, setTour] = useState<Tour | null>(event.tour);
-    const [student, setStudent] = useState(event.student);
+    const {event, isOpen, onClose, onDelete, onDeleteTour, reload} = data;
+    const [tourId, setTourId] = useState(event.tour.id);
+    const [studentId, setStudentId] = useState(event.student.id);
+    const [errorMessages, setErrorMessages] = useState<string[]>([]);
 
     const handleSave = () => {
-        let data = { tour: tour, student: student };
-        onSave(data);
-        onClose();
+        patchStudentOnTour(event.id, tourId, studentId, formatDate(event.start)).then(
+            (_) => {
+                reload(null, null);
```

Comment:
Why reload all the data, can't you just patch the object in the list of studentonTours? Now every time after a drag & drop, the data flickers.

Reply:
It was indeed not needed to reload for a patch or delete, for both this is now directly edited in the schedule instead of using a reload. The flickering is something that will still be happening for the post requests but with the transition of the modal closing this shouldn't be that much of an inconvenience to the user.